backbone/src/main/java/org/researchstack/backbone/ui/ViewTaskActivity.java

```diff
@@ -207,7 +216,11 @@ public void onRequestPermissionsResult(int requestCode,
             if (stepBody instanceof PermissionListener) {
                 ((PermissionListener) stepBody).onPermissionGranted(result);
             }
+        } else if (requestCode == STEP_PERMISSION_LISTENER_REQUEST) {
+            PermissionResult result = new PermissionResult(permissions, grantResults);
+            stepPermissionListener.onPermissionGranted(result);
         }
+        stepPermissionListener = null;
```

Comment:
Because you save this reference when the user calls requestPermissions(listener, permissions) and only for `STEP_PERMISSION_LISTENER_REQUEST`, shouldn't you set this back to null _only_ when the permissionResult is for that particular request? (in practice, this may never really cause an issue, because the screen is limited to "on permission" per screen... but still. :)

What do you think?

Reply:
You're right, I'll move that one line up :)